.ibm/pipelines/openshift-tests.sh

```diff
@@ -0,0 +1,191 @@
+    else
+        echo "Repository ${HELM_REPO_NAME} already exists - updating repository instead."
+    fi
+
+    helm repo update
+}
+
+# install the latest ibmcloud cli on Linux
+install_ibmcloud() {
+  if [[ -x "$(command -v ibmcloud)" ]]; then
+    echo "ibmcloud is already installed."
+  else
+    curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+    echo "the latest ibmcloud cli installed successfully."
+  fi
+}
+
+install_oc() {
+  if [[ -x "$(command -v oc)" ]]; then
+    echo "oc is already installed."
+  else
+    curl -LO https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/linux/oc.tar.gz
+    tar -xf oc.tar.gz
+    mv oc /usr/local/bin/
+    rm oc.tar.gz
+    echo "oc installed successfully."
+  fi
+}
+
+install_helm() {
+  if [[ -x "$(command -v helm)" ]]; then
+    echo "Helm is already installed."
+  else
+    echo "Installing Helm 3 client"
+    mkdir ~/tmpbin && cd ~/tmpbin
+
+    HELM_INSTALL_DIR=$(pwd)
+    curl -sL https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash -f
+    export PATH=${HELM_INSTALL_DIR}:$PATH
+    echo "helm client installed successfully."
+  fi
+}
+
+LOGFILE="pr-${GIT_PR_NUMBER}-openshift-tests-${BUILD_NUMBER}"
+echo "Log file: ${LOGFILE}"
+# source ./.ibm/pipelines/functions.sh
+
+# install ibmcloud
+install_ibmcloud
+
+ibmcloud version
+ibmcloud config --check-version=false
+ibmcloud plugin install -f container-registry
+ibmcloud plugin install -f kubernetes-service
+
+# Using pipeline configuration - environment properties
+ibmcloud login -r "${IBM_REGION}" -g "${IBM_RSC_GROUP}" --apikey "${SERVICE_ID_API_KEY}"
+ibmcloud oc cluster config --cluster "${OPENSHIFT_CLUSTER_ID}"
+
+install_oc
+
+oc version --client
+# oc login -u apikey -p "${SERVICE_ID_API_KEY}" --server="${IBM_OPENSHIFT_ENDPOINT}"
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+echo "$DIR"
+
+oc login --token=${K8S_CLUSTER_TOKEN} --server=${K8S_CLUSTER_URL}
+
+# refresh the name space if exists
+if oc get namespace ${NAME_SPACE} > /dev/null 2>&1; then
+    echo "Namespace ${NAME_SPACE} already exists! refreshing namespace"
+    oc delete namespace ${NAME_SPACE}
+fi
+oc create namespace ${NAME_SPACE}
+
+oc config set-context --current --namespace=${NAME_SPACE}
+
+install_helm
+
+cd $DIR
+
+# Change the namespace of the resources to the one namespace set above
+sed -i "s/namespace:.*/namespace: $NAME_SPACE/g" $DIR/resources/service_account/service-account-rhdh.yaml
+sed -i "s/namespace:.*/namespace: $NAME_SPACE/g" $DIR/resources/cluster_role_binding/cluster-role-binding-k8s.yaml
+sed -i "s/namespace:.*/namespace: $NAME_SPACE/g" $DIR/resources/cluster_role_binding/cluster-role-binding-ocm.yaml
+
+sed -i "s/backstage.io\/kubernetes-id:.*/backstage.io\/kubernetes-id: $K8S_PLUGIN_ANNOTATION/g" $DIR/resources/deployment/deployment-test-app-component.yaml
+
+sed -i "s/GITHUB_APP_APP_ID:.*/GITHUB_APP_APP_ID: $GITHUB_APP_APP_ID/g" $DIR/auth/secrets-rhdh-secrets.yaml
```

Comment:
There are better ways to handle this, like, for example, using customize. But this is ok, for now, we can improve this later.